models/wideresnet.py

```diff
@@ -84,4 +83,93 @@ def forward(self, x):
         out = self.relu(self.bn1(out))
         out = F.avg_pool2d(out, 8)
         out = out.view(-1, self.nChannels)
+        return self.fc(out)
+        
+class WideResNet48(nn.Module):
+    def __init__(self, num_classes, depth=28, widen_factor=2, dropRate=0.0):
+        super(WideResNet48, self).__init__()
+        nChannels = [16, 16*widen_factor, 32*widen_factor, 64*widen_factor]
+        assert((depth - 4) % 6 == 0)
+        n = (depth - 4) / 6
+        block = BasicBlock
+        # 1st conv before any network block
+        self.conv1 = nn.Conv2d(3, nChannels[0], kernel_size=3, stride=1,
+                               padding=1, bias=False)
+        # 1st block
+        self.block1 = NetworkBlock(n, nChannels[0], nChannels[1], block, 1, dropRate, activate_before_residual=True)
+        # 2nd block
+        self.block2 = NetworkBlock(n, nChannels[1], nChannels[2], block, 2, dropRate)
+        # 3rd block
+        self.block3 = NetworkBlock(n, nChannels[2], nChannels[3], block, 2, dropRate)
+        # global average pooling and classifier
+        self.bn1 = nn.BatchNorm2d(nChannels[3], momentum=0.001)
+        self.relu = nn.LeakyReLU(negative_slope=0.1, inplace=True)
+        self.fc = nn.Linear(nChannels[3], num_classes)
+        self.nChannels = nChannels[3]
+
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                n = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
+                m.weight.data.normal_(0, math.sqrt(2. / n))
+            elif isinstance(m, nn.BatchNorm2d):
+                m.weight.data.fill_(1)
+                m.bias.data.zero_()
+            elif isinstance(m, nn.Linear):
+                nn.init.xavier_normal_(m.weight.data)
+                m.bias.data.zero_()
+
+    def forward(self, x):
+        out = self.conv1(x)
+        out = self.block1(out)
+        out = self.block2(out)
+        out = self.block3(out)
+        out = self.relu(self.bn1(out))
+        out = F.avg_pool2d(out, 12)
+        out = out.view(-1, self.nChannels)
+        return self.fc(out)
+        
+class WideResNet96(nn.Module):
+    def __init__(self, num_classes, depth=28, widen_factor=2, dropRate=0.0):
+        super(WideResNet96, self).__init__()
+        nChannels = [16, 16*widen_factor, 32*widen_factor, 64*widen_factor, 64*widen_factor]
```

Comment:
should this line be "nChannels = [16, 16*widen_factor, 32*widen_factor, 64*widen_factor, 128*widen_factor]"?

Reply:
This change doubles the parameter count from 2.67M to 5.93M. I started training the larger model now since I don't remember the performance difference from a few months ago. I'll post the results soon.

Reply:
@chencjGene It doesn't provide any gains. Last layer 64 = max top1 91.8 vs last layer 128 = max top1 91.76. The larger network trains slightly slower and starts to overfit sooner. [wider results](https://pastebin.com/raw/pZ19btH3) [narrower results](https://pastebin.com/raw/YCDCws3m)